### ISSUE TITLE
Superclass mismatch when the class is namespaced

### ISSUE THREAD

**Issue by marceloboeira** [waterlink/mocks.cr#3]
When mocking a  class method inside a module, basically a namespaced class, like:

``` crystal
require "spec"
require "mocks"

module Foo
  class Bar
    def foo
      "bar"
    end
  end
end

create_mock Foo::Bar do
  mock foo
end

describe Foo::Bar do
  it "mocks" do
    instance = Foo::Bar.new
    allow(instance).to receive(foo).and_return("not bar")

    instance.foo.should eq("not bar")
  end
end
```

Instead of creating the mock and returning `not bar`, it is generating the following error/code:

`superclass mismatch for class Foo::Bar (Mocks::BaseDouble for Reference)`

``` crystal
class Foo::Bar 
  @@__mocks_name = "Foo::Bar"

  include ::Mocks::BaseMock
  mock(foo)
end

module ::Mocks
  module InstanceDoubles
    class Foo::Bar < ::Mocks::BaseDouble
      @@name = "Mocks::InstanceDoubles::Foo::Bar"

      def ==(other)
        self.same?(other)
      end

      def ==(other : Value)
        false
      end

      macro mock(method_spec)
        mock({{method_spec}}, nil, ::Foo::Bar.allocate)
      end

      mock(foo)
    end
  end
end
```

**Comment by waterlink** [comment]
Hi @marceloboeira 

Thanks for reporting this bug. I stumbled upon it once too, but quickly used some workaround. Now it looks like it is time to add a test for this and fix it :)

**Comment by waterlink** [comment]
@marceloboeira I have released `0.2.1` version with the fix for this. Could you please upgrade and see if it works for you?

**Comment by marceloboeira** [comment]
@waterlink thanks a lot! 

~~I'll check and get back to you.~~

it is working, thanks again!
